sdk/src/main/java/software/amazon/awssdk/iot/AwsIotMqttConnectionBuilder.java

```diff
@@ -637,14 +640,16 @@ public MqttClientConnection build() {
                 if (config.getPort() != 443) {
                     Log.log(LogLevel.Warn, LogSubject.MqttClient,"Attempting to connect to authorizer with unsupported port. Port is not 443...");
                 }
-                if (tlsOptions.alpnList.size() == 1) {
-                    if (tlsOptions.alpnList.get(0) != "mqtt") {
+                if (config.getUseWebsockets() == false) {
```

Comment:
Why wouldn't we just do this unconditionally?

Reply:
For websockets we do not want to add `mqtt` to the ALPN list. Otherwise the websockets will not be able to make a MQTT connection. Having `mqtt` in the ALPN list was the main thing that was causing custom auth and websockets to not work with MQTT311.